### PR TITLE
Issue #11434 - resolve differences in AliasCheckerSymlinkTests

### LINKED ISSUE #11434: resolve differences in EE9/EE10/Core AliasCheckerSymlinkTests
**Jetty version(s)**
12

**Enhancement Description**
We have `AliasCheckerSymlinkTest` for EE9/EE10 and Jetty Core.

The EE10 version is the most up-to-date with the latest tests, but the Jetty Core and EE9 versions have not been updated.

The EE10 and Jetty Core versions are both testing the same thing so I think we only need one of those.

EE9 checks aliases in the nested `ContextHandler` so I think we still need to keep this one.

### PR DESCRIPTION
**Pull request by lachlan-roberts**
closes #11434

### CODE REVIEW

File: tests/test-integration/src/test/resources/combined/WEB-INF/file [hunk 1]
```diff
@@ -0,0 +1 @@
+This is in the second WEB-INF dir.
```
Comment (gregw) [inline tests/test-integration/src/test/resources/combined/WEB-INF/file:1]: EOLN

File: tests/test-integration/src/test/resources/combined/combinedFile [hunk 2]
```diff
@@ -0,0 +1 @@
+This is a file in the combined resource dir.
```
Comment (gregw) [inline tests/test-integration/src/test/resources/combined/combinedFile:1]: EOLN

File: tests/test-integration/src/test/resources/combined/files/file1 [hunk 3]
```diff
@@ -0,0 +1 @@
+file1 from combined dir
```
Comment (gregw) [inline tests/test-integration/src/test/resources/combined/files/file1:1]: EOLN

File: tests/test-integration/src/test/resources/webroot/files/file2 [hunk 4]
```diff
@@ -0,0 +1 @@
+file1 from webroot
```
Comment (gregw) [inline tests/test-integration/src/test/resources/webroot/files/file2:1]: EOLN

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java [hunk 5]
```diff
@@ -1197,7 +1197,7 @@ protected boolean handleByContextHandler(String pathInContext, ContextRequest re
             return true;
         }
 
-        return super.handleByContextHandler(pathInContext, request, response, callback);
+        return false;
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java]: I don't like this always false return.

I think this method would be better written like:
```java
    protected boolean handleByContextHandler(String pathInContext, ContextRequest request, Response response, Callback callback)
    {
        if (!request.instanceof ServletContextRequest && isProtectedTarget(pathInContext))
            return false;
        return super.handleByContextHandler(pathInContext, request, response, callback);
    }
```

**Review by gregw — CHANGES_REQUESTED** [review]

File: tests/test-integration/src/test/resources/combined/WEB-INF/file (hunk 1 above)
Comment (lachlan-roberts) [reply tests/test-integration/src/test/resources/combined/WEB-INF/file:1]: This is intentional so we don't have to check for newline in the response for the test.

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java (hunk 5 above)
Comment (lachlan-roberts) [reply jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java]: I see what you mean, but this code is incorrect with `isProtectedTarget(pathInContext)` here.

I think this is correct:
```java
boolean initialDispatch = request instanceof ServletContextRequest;
if (!initialDispatch)
    return false;
return super.handleByContextHandler(pathInContext, request, response, callback);
```

**Review by gregw — APPROVED** [review]

**Pull request merged**